### ISSUE TITLE
integrate with Lightning ecosystem CI

### ISSUE THREAD

**Issue by pl-ghost** [microsoft/torchgeo#365]
Hello and so happy to see you use Pytorch-Lightning! :tada: 
Just wondering if you already heard about quite the new **Pytorch Lightning (PL) ecosystem CI** where we would like to invite you to... You can check out our blog post about it: [Stay Ahead of Breaking Changes with the New Lightning Ecosystem CI](https://devblog.pytorchlightning.ai/stay-ahead-of-breaking-changes-with-the-new-lightning-ecosystem-ci-b7e1cf78a6c7) :zap: 
As you use PL framework for your cool project, we would like to enhance your experience and offer you safe updates to our future releases. At this moment, you run tests with a particular PL version, but it may accidentally happen that the next version will be incompatible with your project... :confused: We do not intend to change anything on our project side, but still here we have a solution - ecosystem CI with testing both - your and our latest development head we can find it very early and prevent releasing eventually bad version... :+1: 

**What is needed to do?**
- have some tests, including PL integration
- add config to ecosystem CI - https://github.com/PyTorchLightning/ecosystem-ci

**What will you get?**
- scheduled nightly testing configured for development/stable versions
- slack notification if something went wrong to investigate
- testing also on multi-GPU machine as our gift to you :rabbit: 

cc: @borda

**Comment by adamjstewart** [comment]
Hi @pl-ghost, I took a look at this but it isn't clear to me what benefits Lightning CI would have.

> After a bit of freaking out, you start to troubleshoot what’s going on. Only to discover that some dependency which used to work an hour ago, released a new version with breaking changes.

Wouldn't Lightning CI simply exacerbate this problem? Now instead of our tests starting to fail unexpectedly with every new release, they might start to fail unexpectedly with every new commit.

Unless I'm missing something, I don't see how this would be any different than installing the master branch of Lightning in our existing GitHub Actions CI instead of the latest release.

**Comment by Borda** [comment]
> Wouldn't Lightning CI simply exacerbate this problem? Now instead of our tests starting to fail unexpectedly with every new release, they might start to fail unexpectedly with every new commit.

I think that this is exactly the case, when something starts to fail after release on either side you or we need to make a new release and it is just extra work for something which could be caught much sooner and resolved even before any release...

> Unless I'm missing something, I don't see how this would be any different than installing the master branch of Lightning in our existing GitHub Actions CI instead of the latest release.

yes that is true, but are you doing so, also if you do it it may bring some noise to your development so PRs will be flagged as failing (I mean on the overview when you open details you would see that just integration is failing) even it may not come from the particular change in given PR.

I see you are using the default GH action runners which are CPU only with relatively limited resources... we are offering user multi-GPU machine for let's say bit bigger testing... :rabbit:

**Comment by calebrob6** [comment]
Thanks @Borda! I think this could be a great addition to our CI. I'm interested in creating/running a test that executes a small training run with one of our small datasets and asserts the loss is within some reasonable range. I have a few other things to get off my plate before I can play with this, but will leave the issue open.

**Comment by adamjstewart** [comment]
@calebrob6 are you still planning to work on this or should we close this issue?

**Comment by adamjstewart** [comment]
I think we can close this. Lightning has a new release every week, so we're basically keeping pace with breaking changes without needing additional CI.

**Comment by Borda** [comment]
> I think we can close this. Lightning has a new release every week, so we're basically keeping pace with breaking changes without needing additional CI.

Sure, but how about keeping pace with future releases?
 Like when you develop a new feature it will be already compatible with PL next feature release... :chipmunk:

**Comment by calebrob6** [comment]
Hey Borda! We have pretty low bandwidth and (currently) a long release cycle so I'm not too worried about this. @adamjstewart we could keep this open as a good first issue if someone wants to play with CI
